eng/pipelines/template-tests.yml

```diff
@@ -93,11 +93,12 @@ jobs:
 
       - task: Bash@3
         displayName: Test templates
-        env:             
+        env:
           # Required for Terraform service principal authentication
           ARM_CLIENT_ID: $(arm-client-id)
           ARM_CLIENT_SECRET: $(arm-client-secret)
           ARM_TENANT_ID: $(arm-tenant-id)
+          ARM_SUBSCRIPTION_ID: $(arm-subscription-id)
```

Comment:
I see. I think we will need to come up with a way to perform this test where the failure mode isn't "target another default subscription"

Reply:
What was happening under the covers is that a recent change to `azd` changed such that the default subscription is stored on a config file. When a subscription isn't explicitly specified, `azd` should fail. However, due to a different issue #1024 with the implementation this wasn't true in some cases. That caused the default subscription to be persisted in our `daily` runs, and subsequent runs to reuse the default.

I think the change is to avoid defaults entirely and specify the subscription ID explicitly to `azd`. I've made a change in #1043 and it is working as expected.